### PR TITLE
Fixed a few tests that didn't assert on the outcome

### PR DESCRIPTION
**Pull request by bart-vmware**
While looking at warnings from Resharper, I found some tests that didn't assert on the outcome.

After correcting these tests, some of them failed. I've adapted the target logic accordingly to make the tests succeed without changing their assumptions.

Note this is a breaking change. If taking a breaking change is not an option, we should adapt the tests instead to match the current behavior.

### CODE REVIEW

File: src/Common/src/Common/Util/MimeType.cs [hunk 1]
```diff
@@ -144,7 +144,7 @@ protected void CheckParameters(string attribute, string value)
             if (PARAM_CHARSET.Equals(attribute))
             {
                 value = Unquote(value);
-                Encoding.GetEncoding(value);
+                _ = Encoding.GetEncoding(value);
```
Comment (hananiel) [inline src/Common/src/Common/Util/MimeType.cs:147]: Probably ignore this suggestion - I dont see value ...

**Review by hananiel — APPROVED** [review]
LGTM. See other comment.

File: src/Common/src/Common/Util/MimeType.cs (hunk 1 above)
Comment (bart-vmware) [reply src/Common/src/Common/Util/MimeType.cs:147]: This suppresses the R# warning that the return value of a pure function is unused, which is usually an indication of a mistake. In this case however, I believe the try/catch block is used to detect failure, whereas the actual returned encoding is irrelevant here.

Adding the `_` (discard) expresses that ignoring the return value is intentional. By being explicit about that, I believe this improves code readability: Future readers won't have to wonder whether this was intentional.

Based on the above, are you okay with keeping this?

Comment (hananiel) [reply src/Common/src/Common/Util/MimeType.cs:147]: Ok sounds good.

File: src/Management/test/EndpointBase.Test/Metrics/Observer/HttpClientDesktopObserverTest.cs [hunk 2]
```diff
@@ -74,10 +74,11 @@ public void GetTagContext_ReturnsExpected()
 
             var req = GetHttpRequestMessage();
             var labels = observer.GetLabels(req, HttpStatusCode.InternalServerError).ToList();
-            labels.Contains(KeyValuePair.Create("clientName", (object)"localhost:5555"));
-            labels.Contains(KeyValuePair.Create("uri", (object)"/foo/bar"));
-            labels.Contains(KeyValuePair.Create("status", (object)"500"));
-            labels.Contains(KeyValuePair.Create("method", (object)"GET"));
+
+            Assert.Contains(KeyValuePair.Create("clientName", (object)"localhost:5555"), labels);
+            Assert.Contains(KeyValuePair.Create("uri", (object)"/foo/bar"), labels);
```
Comment (hananiel) [inline src/Management/test/EndpointBase.Test/Metrics/Observer/HttpClientDesktopObserverTest.cs:79]: This is the correct assertion. And the code change is correct.

**Comment by hananiel** [comment]
> Note this is a breaking change. If taking a breaking change is not an option, we should adapt the tests instead to match the current behavior.
This fixes the uri to be compatible with the way spring-boot apps report uri. So shouldnt be a problem. Go ahead and merge.

**Pull request merged**